modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/util/DDMStructureUtil.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.headless.delivery.internal.dto.v1_0.util;
+
+import com.liferay.dynamic.data.mapping.model.DDMStructure;
+import com.liferay.dynamic.data.mapping.service.DDMStructureService;
+import com.liferay.headless.delivery.dto.v1_0.StructuredContent;
+import com.liferay.portal.kernel.security.auth.PrincipalException;
+
+import javax.ws.rs.ForbiddenException;
+
+/**
+ *@author Luis Miguel Barcos
+ */
+public class DDMStructureUtil {
+
+	public static DDMStructure getDDMStructure(
+			DDMStructureService ddmStructureService,
+			StructuredContent structuredContent)
+		throws Exception {
+
+		try {
+			return ddmStructureService.getStructure(
```

Comment:
mmm encapsulating this seems to much. Ideally it would be a exceptcionMapper but I don't think it can be done because the exception is too generic. I would leave this way but I don't know if Brian is going to change it.

Reply:
We already have a PrincipalExceptionMapper in Vulcan, that maps to NOT FOUND in get requests and to FORBIDDEN in other cases. Maybe we can use it, so we have a uniform behavior with other endpoints

https://github.com/liferay/liferay-portal/blob/master/modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/exception/mapper/PrincipalExceptionMapper.java

Reply:
Finally, I deleted the DDMStructureUtil and use the Principal ExceptionMapper. I was looking for the reason because we have this in this way but I didn't find anything. Just [this LPS](https://issues.liferay.com/browse/LPS-93617) where was added.